### PR TITLE
EventsController changes

### PR DESCRIPTION
**Pull request by Luuxo77**
- Endpoints for adding rating, interest in event and participate in event
- Rework and add filters for getting a list of events
- Add more data to event dtos

### CODE REVIEW

File: Api/Controllers/EventsController.cs [hunk 1]
```diff
@@ -74,63 +67,132 @@ public ActionResult DeleteEvent(Guid id)
...
+            if (participants.Count() == 1)
+            {
+                events = participants.First() switch
+                {
+                    ParticipantsType.To50 => events.Where(x => x.Participants.Count <= 50 && x.Participants.Count >= 0),
+                    ParticipantsType.From50To100 => events.Where(x => x.Participants.Count >= 50 && x.Participants.Count <= 100),
+                    ParticipantsType.Above100 => events.Where(x => x.Participants.Count >= 100),
+                    _ => throw new InvalidDomainEnumException()
+                };
+            }
```
Comment (WildJellybear) [inline Api/Controllers/EventsController.cs:152]: Is it necessarry to consider case for only one requirement separately?

File: Api/Controllers/EventsController.cs [hunk 2]
```diff
@@ -74,63 +67,132 @@ public ActionResult DeleteEvent(Guid id)
...
+            if (time.Count() == 1)
+            {
+                events = time.First() switch
+                {
+                    TimeType.Past => events.Where(x => x.EndDate <= DateTime.Now),
+                    TimeType.Current => events.Where(x => x.StartDate <= DateTime.Now && x.EndDate >= DateTime.Now),
+                    TimeType.Future => events.Where(x => x.StartDate >= DateTime.Now),
+                    _ => throw new InvalidDomainEnumException()
+                };
+            }
```
Comment (WildJellybear) [inline Api/Controllers/EventsController.cs:176]: Same as comment above

**Review by WildJellybear — APPROVED** [review]
I cannot check changes in migrations, however everythong else looks ok

File: Api/Controllers/EventsController.cs (hunk 1 above)
Comment (WildJellybear) [inline Api/Controllers/EventsController.cs]: Is it necessarry to consider case for only one requirement separately?

File: Api/Controllers/EventsController.cs (hunk 2 above)
Comment (WildJellybear) [inline Api/Controllers/EventsController.cs]: Same as comment above

File: Api/Controllers/EventsController.cs (hunk 1 above)
Comment (Luuxo77) [reply Api/Controllers/EventsController.cs]: I wrote it just to work, it is not necessary but if I didn't consider it, there would be more if's below. I will simplify it in next commit.

**Review by Patonymous — APPROVED** [review]
LGTM

**Pull request merged**